### PR TITLE
Use Kernnetz more flexible

### PR DESCRIPTION
**Pull request by JulianGeis**
Before asking for a review for this PR make sure to complete the following checklist:

- [x] Workflow with target rule `ariadne_all` completes without errors
- [x] The logic of `export_ariadne_variables` has been adapted to the changes
- [x] One or several figures that validate the changes in the PR have been posted as a comment
- [x] A brief description of the changes has been added to `Changelog.md`
- [x] The latest `main` has been merged into the PR
- [x] The config has a new prefix of the format `YYYYMMDDdescriptive_title`

### CODE REVIEW

**Comment by JulianGeis** [comment]
I added the latest data for the Kernnetz lines according to the "final Proposal" of the FNB.

**Before clustering**

All Kernnetz pipes:
image

All IPCEI and PCI Projects:
image

Kernnetz according to FNB:
- some lines are wrong e.g. the North South link in Mecklenburg-Vorpommern, which is correctly not plotted before
image

In the config you now have those filtering options:
  **ipcei_pci_only: bool** -> only model the pipes of important interest
  **cutoff_year: int* -> model the Kernnetz only until a specific year
  **force_all_ipcei_pci: bool** -> model the "regular" Kernnetz only until a specific year but force the ipcei and pci projects also after the cutoff

and this
  aggregate_build_years: "min" to specify an aggregation strategy of the parallel pipes (min, mean, max)

Retrofitting costs:
The costs of the Kernnetz pipes are modelled as follows: If there is only one pipe determined then use the info from FNB if it is retrofitted or not. If you have to aggregate several parallel pipes, then the aggregated pipe is considered as retrofitted and the retrofitting costs are assumed if more than 60% of the parallel pipes are retrofitted. This is no config param but can be adjusted in build_wasserstoff_kernnetz.py

File: workflow/Snakefile [hunk 1]
```diff
@@ -412,13 +412,18 @@ use rule build_population_weighted_energy_totals from pypsaeur with:
 rule build_wasserstoff_kernnetz:
     params:
         reload_locations=config_provider("wasserstoff_kernnetz", "reload_locations"),
```
Comment (toniseibold) [inline workflow/Snakefile:414]: only pass on the kernnetz params?

**Review by toniseibold — COMMENTED** [review]
Only minor changes necessary.
Would be interesting to have a higher clustering resolution to make visual comparison between FNB and PyPSA results easier?

File: workflow/Snakefile (hunk 1 above)
Comment (JulianGeis) [reply workflow/Snakefile:414]: makes sense, done

File: config/config.yaml [hunk 2]
```diff
@@ -246,6 +246,10 @@ wasserstoff_kernnetz:
   reload_locations: false
```
[the remark this answers is not in the document]
Comment (JulianGeis) [reply config/config.yaml:246]: for me it did run trough. :) Geocoding is kind of random sometimes. However this config setting should be false per default and the function is only used if the data has to be updated. So there is some manual work involved anyways to clean the data. So I guess it is fine to keep is like it is?

File: workflow/scripts/build_wasserstoff_kernnetz.py [hunk 3]
```diff
@@ -373,6 +435,7 @@ def assign_locations(df, locations):
     wasserstoff_kernnetz = load_and_merge_raw(
         snakemake.input.wasserstoff_kernnetz_1,
         snakemake.input.wasserstoff_kernnetz_2,
+        snakemake.input.wasserstoff_kernnetz_3,
     )
 
     wasserstoff_kernnetz = prepare_dataset(wasserstoff_kernnetz)
```
[the remark this answers is not in the document]
Comment (JulianGeis) [reply workflow/scripts/build_wasserstoff_kernnetz.py:441]: done

**Pull request merged**